saleor/plugins/manager.py

```diff
@@ -425,8 +426,21 @@ def calculate_order_line_total(
         order_line: "OrderLine",
         variant: "ProductVariant",
         product: "Product",
+        lines: Iterable["OrderLine"],
     ) -> OrderTaxedPricesData:
-        default_value = base_order_calculations.base_order_line_total(order_line)
+        base_subtotal = base_order_subtotal(order, lines)
+        subtotal, shipping_price = propagate_order_discount_on_order_prices(
+            order, lines
+        )
```

Comment:
With this implementation we fire `propagate_order_discount_on_order_prices` function for every single order line. The function call db, so it is not an optimal solution. Maybe we can pass `subtotal` and `shipping price` in args?

Reply:
It is okay. All db data is prefetched. I assumed, that calculating the values inside these method will allow to be "independent" method. Just to confirm I've made additional test when you're calling fetch_prices logic:
```
In [94]: def a():
    ...:     o = Order.objects.filter(currency="USD")[4]
    ...:     fetch_order_prices_if_expired(o, manager, o.lines.all())
    ...:

In [95]: a()
Without propagation
Without propagation
Without propagation
Number of queries 17

In [96]: a()
With propagation
With propagation
With propagation
Number of queries 17
```
Where "Without" is the logic with commented part of calculation inside the manager. 
"With" is the logic from this PR. 
Both results end up with 17db queries. Tested on order with 3 lines.